gensim/models/doc2vec.py

```diff
@@ -635,8 +635,13 @@ def scan_vocab(self, documents, progress_per=10000, trim_rule=None):
         min_reduce = 1
         interval_start = default_timer() - 0.00001  # guard against next sample being identical
         interval_count = 0
+        checked_string_types = 0
         vocab = defaultdict(int)
         for document_no, document in enumerate(documents):
+            if not checked_string_types:
+                if isinstance(document.words, string_types):
+                    warnings.warn("'words' should be a list of unicode strings. True string type provided.")
```

Comment:
Instead of "True string type", why not output the actual type of `document.words`?

Plus maybe even output the entire `document` example, for explicitness and clarity.